test/functional/shada/registers_spec.lua

```diff
@@ -160,4 +160,16 @@ describe('ShaDa support code', function()
     reset()
     eq({{'\171'}, 'v'}, getreg('e'))
   end)
+
+  it('remembers which register was the unnamed register when loading',
+  function()
+    setreg('0', {'zero'}, 'c')
+    setreg('1', {'one'}, 'cu')
+    setreg('2', {'two'}, 'c')
+    nvim_command('qall')
+    reset()
+    eq({{'zero'}, 'v'}, getreg('0'))
+    eq({{'one'}, 'v'}, getreg('1'))
+    eq({{'one'}, 'v'}, getreg('"'))
+  end)
```

Comment:
The point is that you check what would be done if there are some registers stored in a ShaDa file, but among them there is no one linked with unnamed register. This is achieved by setting `'3'` without setting anything else. Or setting `'a'`. Better set both though.